INDI/INDI/indiserver.c

```diff
@@ -1119,109 +1113,152 @@ static void newFIFO(void)
             else
+            {
                 startRemoteDvr(dp);
+            }
         }
         else
         {
-            for (dp = dvrinfo; dp < &dvrinfo[ndvrinfo]; dp++)
+            for (dp = dvrinfo; dp < (dvrinfo + ndvrinfo); dp++)
             {
-                fprintf(stderr, "%s: dp->name: %s - tDriver: %s\n", ts, dp->name, tDriver);
-                if (!strcmp(dp->name, tDriver) && dp->active == 1)
-                {
-                    fprintf(stderr, "%s: name: %s - dp->dev[0]: %s\n", ts, tName, dp->dev[0]);
-
-                    /* If device name is given, check against it before shutting down */
-                    //if (tName[0] && strcmp(dp->dev[0], tName))
-                    if (tName[0] && isDeviceInDriver(tName, dp) == 0)
-                        continue;
-                    if (verbose)
-                        fprintf(stderr, "%s: FIFO: Shutting down driver: %s\n", ts, tDriver);
-
-                    //                    for (i = 0; i < dp->ndev; i++)
-                    //                    {
-                    //                        /* Inform clients that this driver is dead */
-                    //                        XMLEle *root = addXMLEle(NULL, "delProperty");
-                    //                        addXMLAtt(root, "device", dp->dev[i]);
-
-                    //                        prXMLEle(stderr, root, 0);
-                    //                        Msg *mp = newMsg();
-
-                    //                        q2Clients(NULL, 0, dp->dev[i], NULL, mp, root);
-                    //                        if (mp->count > 0)
-                    //                            setMsgXMLEle(mp, root);
-                    //                        else
-                    //                            freeMsg(mp);
-                    //                        delXMLEle(root);
-                    //                    }
-
-                    shutdownDvr(dp, 0);
-                    break;
-                }
+                fprintf(stderr, "%s: Looking for driver to stop dp->name[%s] ==? tDriver[%s]\n", ts, dp->name, tDriver);
+                if (strcmp(dp->name, tDriver) || !dp->active) { continue; }
+                fprintf(stderr, "%s: Found driver to stop:  tName: [%s] - dp->dev[0]: [%s]\n"
+                       , ts
+                       , tName?tName:"<null>"
+                       , (dp->ndev>0 && dp->dev[0]) ? dp->dev[0] : "<null>"
+                       );
+
+                removeDvrFromRestartList(dp);
+
+                /* If device name is given, check against it before shutting down */
+                if (tName[0] && isDeviceInDriver(tName, dp) == 0) { continue; }
+                if (verbose) { fprintf(stderr, "%s: FIFO: Shutting down driver: %s\n", ts, tDriver); }
+                shutdownDvr(dp, 0);
+                if (verbose) { fprintf(stderr, "%s: FIFO: Driver Shut down complete: %s\n", ts, tDriver); }
+                break;
```

Comment:
This new code ensures a stop command via the FIFO interacts correctly with a driver that may be wating to restart.